homeassistant/components/smartthings/light.py

```diff
@@ -0,0 +1,204 @@
+"""
+Support for lights through the SmartThings cloud API.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/smnartthings.light/
+"""
+import asyncio
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, ATTR_COLOR_TEMP, ATTR_HS_COLOR, ATTR_TRANSITION,
+    SUPPORT_BRIGHTNESS, SUPPORT_COLOR, SUPPORT_COLOR_TEMP, SUPPORT_TRANSITION,
+    Light)
+import homeassistant.util.color as color_util
+
+from . import SmartThingsEntity
+from .const import DATA_BROKERS, DOMAIN
+
+DEPENDENCIES = ['smartthings']
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Platform uses config entry setup."""
+    pass
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add lights for a config entry."""
+    broker = hass.data[DOMAIN][DATA_BROKERS][config_entry.entry_id]
+    async_add_entities(
+        [SmartThingsLight(device) for device in broker.devices.values()
+         if is_light(device)], True)
+
+
+def is_light(device):
+    """Determine if the device should be represented as a light."""
+    from pysmartthings import Capability
+
+    # Must be able to be turned on/off.
+    if Capability.switch not in device.capabilities:
+        return False
+    # Not a fan (which might also have switch_level)
+    if Capability.fan_speed in device.capabilities:
+        return False
+    # Must have one of these
+    light_capabilities = [
+        Capability.color_control,
+        Capability.color_temperature,
+        Capability.switch_level
+    ]
+    if any(capability in device.capabilities
+           for capability in light_capabilities):
+        return True
+    return False
+
+
+def convert_scale(value, value_scale, target_scale, round_digits=4):
+    """Convert a value to a different scale."""
+    return round(value * target_scale / value_scale, round_digits)
+
+
+class SmartThingsLight(SmartThingsEntity, Light):
+    """Define a SmartThings Light."""
+
+    def __init__(self, device):
+        """Initialize an SmartThingsLight."""
+        SmartThingsEntity.__init__(self, device)
```

Comment:
There's no init in the light or entity classes.

Reply:
There isn't today, but if ever added, this code would break.  We'd never want to pass `device` to the Light hierarchy anyway.  This seems to make the code unnecessarily brittle?

Reply:
If there's an init in another parent class in the future, `SmartThingsEntity` will also have to call super and this code won't break.

Reply:
Changed in latest commit!  All convos resolved.